src/iop/colorbalancergb.c

```diff
@@ -942,30 +1277,81 @@ void gui_init(dt_iop_module_t *self)
+
+  g->midtones_C = dt_bauhaus_slider_from_params(self, "midtones_C");
+  dt_bauhaus_slider_set_soft_range(g->midtones_C, 0., 0.02);
+  dt_bauhaus_slider_set_step(g->midtones_C, 0.005);
+  dt_bauhaus_slider_set_digits(g->midtones_C, 4);
+  dt_bauhaus_slider_set_factor(g->midtones_C, 100.0f);
+  dt_bauhaus_slider_set_format(g->midtones_C, "%.2f %%");
+  gtk_widget_set_tooltip_text(g->midtones_C, _("chroma of the color exponent in midtones"));
+
   // Page masks
   self->widget = dt_ui_notebook_page(g->notebook, _("masks"), _("isolate luminances"));
 
+  gtk_box_pack_start(GTK_BOX(self->widget), dt_ui_section_label_new(_("luminance ranges")), FALSE, FALSE, 0);
+
+  g->area = GTK_DRAWING_AREA(dtgtk_drawing_area_new_with_aspect_ratio(0.75));
+  g_signal_connect(G_OBJECT(g->area), "draw", G_CALLBACK(dt_iop_tonecurve_draw), self);
+  gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->area), FALSE, FALSE, 0);
+
   g->shadows_weight = dt_bauhaus_slider_from_params(self, "shadows_weight");
   dt_bauhaus_slider_set_digits(g->shadows_weight, 4);
   dt_bauhaus_slider_set_step(g->shadows_weight, 0.1);
   dt_bauhaus_slider_set_format(g->shadows_weight, "%.2f %%");
   dt_bauhaus_slider_set_factor(g->shadows_weight, 100.0f);
   gtk_widget_set_tooltip_text(g->shadows_weight, _("weight of the shadows over the whole tonal range"));
-
-  g->midtones_weight = dt_bauhaus_slider_from_params(self, "midtones_weight");
-  dt_bauhaus_slider_set_soft_range(g->midtones_weight, -2., +2.);
-  dt_bauhaus_slider_set_step(g->midtones_weight, 0.1);
-  dt_bauhaus_slider_set_digits(g->midtones_weight, 4);
-  dt_bauhaus_slider_set_format(g->midtones_weight, "%.2f EV");
-  gtk_widget_set_tooltip_text(g->midtones_weight, _("peak white luminance value used to normalize the power function"));
+  dt_bauhaus_widget_set_quad_paint(g->shadows_weight, dtgtk_cairo_paint_showmask,
+                                   CPF_STYLE_FLAT | CPF_DO_NOT_USE_BORDER, NULL);
+  dt_bauhaus_widget_set_quad_toggle(g->shadows_weight, TRUE);
+  g_signal_connect(G_OBJECT(g->shadows_weight), "quad-pressed", G_CALLBACK(mask_callback), self);
+
+  g->grey_fulcrum = dt_bauhaus_slider_from_params(self, "grey_fulcrum");
+  dt_bauhaus_slider_set_digits(g->grey_fulcrum, 4);
+  dt_bauhaus_slider_set_step(g->grey_fulcrum, 0.01);
+  dt_bauhaus_slider_set_format(g->grey_fulcrum, "%.2f %%");
+  dt_bauhaus_slider_set_factor(g->grey_fulcrum, 100.0f);
+  gtk_widget_set_tooltip_text(g->grey_fulcrum, _("position of the middle-grey reference for masking."));
+  dt_bauhaus_widget_set_quad_paint(g->grey_fulcrum, dtgtk_cairo_paint_showmask,
+                                   CPF_STYLE_FLAT | CPF_DO_NOT_USE_BORDER, NULL);
+  dt_bauhaus_widget_set_quad_toggle(g->grey_fulcrum, TRUE);
+  g_signal_connect(G_OBJECT(g->grey_fulcrum), "quad-pressed", G_CALLBACK(mask_callback), self);
+  gtk_box_pack_start(GTK_BOX(self->widget), g->grey_fulcrum, FALSE, FALSE, 0);
```

Comment:
You need to remove this line otherwise we get:
`(darktable:1419180): Gtk-CRITICAL **: 08:15:40.228: gtk_box_pack: assertion '_gtk_widget_get_parent (child) == NULL' failed`

Reply:
fixed